cypress/integration/6-crypto/crypto.spec.ts

```diff
@@ -60,13 +58,21 @@ describe("Cryptography", () => {
                 roomId => cy.inviteUser(roomId, bot.getUserId())
                     .then(() => cy.visit("/#/room/" + roomId))
                     .then(
-                        () => cy.wrap(
-                            new Promise<void>(
-                                resolve => waitForRoomEncryption(bot, roomId, resolve),
-                            ).then(() => bot.sendMessage(roomId, {
-                                body: "Top secret message",
-                                msgtype: "m.text",
-                            })),
+                        () => cy.window().then(
+                            win => cy.wrap(
+                                new Promise<void>(
+                                    resolve => waitForEvent(
+                                        bot,
+                                        win.matrixcs.RoomStateEvent.Update,
+                                        // @ts-ignore - protected
+                                        () => bot.roomList.isRoomEncrypted(roomId),
+                                        resolve,
+                                    ),
+                                ).then(() => bot.sendMessage(roomId, {
+                                    body: "Top secret message",
+                                    msgtype: "m.text",
+                                })),
+                            ),
                         ),
                     ),
```

Comment:
this doesn't seem to match the code-style for the rest of the codebase, see https://github.com/vector-im/element-meta/pull/216 L351 for guidance

Reply:
Yeah, I'm not thrilled with it either; none of the options here seem especially great. I could declare variables outside the Cypress chainables with `let` and then assign them, but the Cypress docs seem pretty emphatically opposed to that. I could use `this` to pull in all of the shared context at once and not have to do as much nesting, but that doesn't type-check. Or I could have this excessive nesting :/ What do you think is a better way to organize this?

Reply:
We normally keep the `() =>` arrow func on the same line as the `.then` this removes a whole layer of nesting